extensions/windows-event-log/tests/BookmarkTests.cpp

```diff
@@ -40,7 +40,10 @@ std::unique_ptr<Bookmark> createBookmark(TestPlan &test_plan,
                                          const utils::Identifier &uuid = IdGenerator::getIdGenerator()->generate()) {
   const auto state_manager = test_plan.getStateManagerProvider()->getCoreComponentStateManager(uuid);
   const auto logger = test_plan.getLogger();
-  return utils::make_unique<Bookmark>(channel, L"*", "", uuid, false, state_manager, logger);
+  state_manager->beginTransaction();
+  auto bookmark = utils::make_unique<Bookmark>(channel, L"*", "", uuid, false, state_manager, logger);
+  state_manager->commit();
```

Comment:
Could we somehow make this easier to use correctly and hard/impossible to use incorrectly? I'm thinking about transaction objects having state mutating capabilities, and the state manager only handing them out. onTrigger could get one as an argument.

Reply:
If you want to make a change to the state (set or clear) without being in a transaction, your call will fail, returning false.
I don't think it is possible to auto-commit outside of onTrigger. It is the user's responsibility to decide when/how often/if to commit or rollback. I cannot think of a way that is universally applicable everywhere.
Transaction objects wouldn't solve this problem, and they would break preexisting interface of state manager, affecting all users, even in onTriggers.

Reply:
A conclusion was made that state changes would be autocommit (SQL-like) when outside of a transaction (such as in onTrigger). A transaction would begin and at the end of the call it would finish with commit. If users want to, they can still begin and commit/rollback a transaction manually if they wish so.